examples/xml-operations/xml_operations.bal

```diff
@@ -30,23 +30,32 @@ public function main() returns error? {
     // `x[i]` gives i-th item (empty sequence if none).
     io:println(x3[0]);
 
-    // `x.id` accesses required attribute named `id`:
-    // result is `error` if there is no such attribute
-    // or if `x` is not a singleton.
+    // `x.id` accesses a required attribute named `id`: result is `error` if there is no such 
+    // attribute or if `x` is not a singleton.
     xml x5 = xml `<para id="greeting">Hello</para>`;
     string id = check x5.id;
 
+    // Since an attribute named `id` exists in the `xml` value on which required attribute access 
+    // is done the result of the access will be the value of the attribute (`"greeting"`).
     io:println(id);
 
-    // `x?.id` accesses optional attribute named `id`:
-    // result is `()` if there is no such attribute.
+    // `x?.id` accesses an optional attribute named `id`: result is `()` if there is no such 
+    // attribute.
     string? name = check x5?.name;
 
+    // Since an attribute named `name` does not exist in the `xml` value on which optional 
+    // attribute access is done the result of the access is nil. Therefore the following
+    // `is` check evaluates to `true`.
     io:println(name is ());
 
     // Mutate an element using `e.setChildren(x)`.
     x2.setChildren(xml `<language>French</language>`);
 
+    // The `xml` value assigned to `x2` will now include `<language>French</language>` as a 
+    // child element.
     io:println(x2);
+
+    // Since the value assigned to `x3` used `x2` in the concatenation, the change will also 
```

Comment:
```suggestion
    // Since the value assigned to `x3` used `x2` in the concatenation, `x2.setChildren` will also change `x3`
```

Reply:
Used "reflected in it" because it is basically the same value that changes. And `x3` having a reference to the same value is why the change is reflected in `x3`.

Reply:
let me try again

Reply:
Resolving this